src/RequestHandler/inprocess/inprocessapplication.cpp

```diff
@@ -564,15 +623,16 @@ IN_PROCESS_APPLICATION::ExecuteApplication(
                 ASPNETCORE_EVENT_INPROCESS_THREAD_EXIT,
                 strEventMsg.QueryStr());
         }
-       
-        // error. the thread exits after application started
-        // Question: should we shutdown current worker process or keep the application in failure state?
-        // for now, we reccylce to keep the same behavior as that of out-of-process
-        if (m_fManagedAppLoaded)
-        {
-            Recycle();
-        }
+
+        //
+        // No matter what we need to cause recycle to be called on the worker process. We also want to remove the application from the application
+        // manager as by dereferencing it in the mananger will trigger recycle. We will post a notify configuration change here, allowing us to receive
+        // a global notification event, which would eventually cause the application to be dereferenced. This is only
+        // done if shutdown has not been called from native.
+        //
+        g_pHttpServer->NotifyConfigurationChange(m_pConfig->QueryConfigPath()->QueryStr());
```

Comment:
why you want to treat this error as configuration change? nothing changed with configuration.

Reply:
We need to remove the application from the application manager somehow, such that we can recycle the worker process eventually.

Reply:
I may be able to change the logic to call Recycle here instead, however I am not sure how to propagate that the server has shutdown to the ApplicationManager.